iothub/device/src/Authentication/ClientAuthenticationWithSakRefresh.cs

```diff
@@ -26,8 +46,78 @@ internal class ClientAuthenticationWithSakRefresh : ClientAuthenticationWithToke
                 sasTokenTimeToLive,
                 sasTokenRenewalBuffer)
         {
-            _sharedAccessKey = sharedAccessKey ?? throw new ArgumentNullException(nameof(sharedAccessKey));
-            _sharedAccessKeyName = sharedAccessKeyName;
+            Argument.AssertNotNullOrWhiteSpace(sharedAccessKey, nameof(sharedAccessKey));
+            SharedAccessKey = sharedAccessKey;
+
+            if (sharedAccessKeyName != null
+                && sharedAccessKeyName.IsNullOrWhiteSpace())
+            {
+                throw new ArgumentException("Shared access key name cannot be white space.");
+            }
+
+            SharedAccessKeyName = sharedAccessKeyName;
+        }
+
+        /// <summary>
+        /// Creates an instance of this class.
+        /// </summary>
+        /// <param name="connectionString">
+        /// The connection string containing the device Id, optional module Id, shared access key name
+        /// and shared access key to be used for authenticating with IoT hub service.
+        /// </param>
+        /// <param name="sasTokenTimeToLive">
+        /// The suggested time to live value for the generated SAS tokens.
+        /// The default value is 1 hour.
+        /// </param>
+        /// <param name="sasTokenRenewalBuffer">
+        /// The time buffer before expiry when the token should be renewed, expressed as a percentage of the time to live.
+        /// The default behavior is that the token will be renewed when it has 15% or less of its lifespan left.
+        /// </param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="connectionString"/> is null.</exception>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="connectionString"/> is empty or whitespace.</exception>
+        public ClientAuthenticationWithSakRefresh(
```

Comment:
this is where we consolidate the old class `ClientAuthenticationWithConnectionString.cs`